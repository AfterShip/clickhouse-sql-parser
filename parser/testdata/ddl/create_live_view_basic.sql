CREATE LIVE VIEW my_live_view
WITH TIMEOUT 10 TO my_destination(id String)
AS SELECT id FROM my_table;
