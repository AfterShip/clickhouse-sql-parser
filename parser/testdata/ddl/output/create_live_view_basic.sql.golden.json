[
  {
    "CreatePos": 0,
    "StatementEnd": 101,
    "Name": {
      "Database": null,
      "Table": {
        "Name": "my_live_view",
        "QuoteType": 1,
        "NamePos": 17,
        "NameEnd": 29
      }
    },
    "IfNotExists": false,
    "UUID": null,
    "OnCluster": null,
    "Destination": {
      "ToPos": 46,
      "TableIdentifier": {
        "Database": null,
        "Table": {
          "Name": "my_destination",
          "QuoteType": 1,
          "NamePos": 49,
          "NameEnd": 63
        }
      },
      "TableSchema": null
    },
    "TableSchema": {
      "SchemaPos": 63,
      "SchemaEnd": 73,
      "Columns": [
        {
          "NamePos": 64,
          "ColumnEnd": 73,
          "Name": {
            "Ident": {
              "Name": "id",
              "QuoteType": 1,
              "NamePos": 64,
              "NameEnd": 66
            },
            "DotIdent": null
          },
          "Type": {
            "Name": {
              "Name": "String",
              "QuoteType": 1,
              "NamePos": 67,
              "NameEnd": 73
            }
          },
          "NotNull": null,
          "Nullable": null,
          "DefaultExpr": null,
          "MaterializedExpr": null,
          "AliasExpr": null,
          "Codec": null,
          "TTL": null,
          "Comment": null,
          "CompressionCodec": null
        }
      ],
      "AliasTable": null,
      "TableFunction": null
    },
    "WithTimeout": {
      "WithTimeoutPos": 30,
      "Expr": null,
      "Number": {
        "NumPos": 43,
        "NumEnd": 45,
        "Literal": "10",
        "Base": 10
      }
    },
    "SubQuery": {
      "HasParen": false,
      "Select": {
        "SelectPos": 78,
        "StatementEnd": 101,
        "With": null,
        "Top": null,
        "HasDistinct": false,
        "SelectItems": [
          {
            "Expr": {
              "Name": "id",
              "QuoteType": 1,
              "NamePos": 85,
              "NameEnd": 87
            },
            "Modifiers": [],
            "Alias": null
          }
        ],
        "From": {
          "FromPos": 88,
          "Expr": {
            "Table": {
              "TablePos": 93,
              "TableEnd": 101,
              "Alias": null,
              "Expr": {
                "Database": null,
                "Table": {
                  "Name": "my_table",
                  "QuoteType": 1,
                  "NamePos": 93,
                  "NameEnd": 101
                }
              },
              "HasFinal": false
            },
            "StatementEnd": 101,
            "SampleRatio": null,
            "HasFinal": false
          }
        },
        "ArrayJoin": null,
        "Window": null,
        "Prewhere": null,
        "Where": null,
        "GroupBy": null,
        "WithTotal": false,
        "Having": null,
        "OrderBy": null,
        "LimitBy": null,
        "Limit": null,
        "Settings": null,
        "Format": null,
        "UnionAll": null,
        "UnionDistinct": null,
        "Except": null
      }
    }
  }
]